File: verdict-backend/src/services/nia.ts
/**
 * Nia API — document indexing & RAG search (ref: IMPLEMENTATION_PLAN Step 2.2, BACKEND_STRUCTURE)
 */
import axios from 'axios';

const nia = axios.create({
  baseURL: process.env.NIA_BASE_URL,
  headers: { Authorization: `Bearer ${process.env.NIA_API_KEY}` },
  timeout: 10_000,
});

export async function indexDocument(params: {
  indexId: string;
  documentId: string;
  content: string;
  metadata?: Record<string, unknown>;
}) {
  const { data } = await nia.post('/index', params);
  return data;
}

export async function searchIndex(params: {
  indexId: string;
  query: string;
  topK?: number;
  filters?: Record<string, unknown>;
}): Promise<Array<{ id: string; content: string; score: number; metadata?: Record<string, unknown> }>> {
  const { data } = await nia.post('/search', {
    ...params,
    topK: params.topK ?? 5,
  });
  return data.results ?? [];
}

File: verdict-backend/app/services/nia.py
import httpx
from app.config import settings


async def index_document(index_id: str, document_id: str, content: str, metadata: dict = None):
    async with httpx.AsyncClient(
        base_url=settings.NIA_BASE_URL,
        headers={"Authorization": f"Bearer {settings.NIA_API_KEY}"},
        timeout=10.0,
    ) as client:
        resp = await client.post("/index", json={
            "indexId": index_id,
            "documentId": document_id,
            "content": content,
            "metadata": metadata or {},
        })
        resp.raise_for_status()
        return resp.json()


async def search_index(index_id: str, query: str, top_k: int = 5, filters: dict = None) -> list[dict]:
    if not settings.NIA_API_KEY:
        return []
    try:
        async with httpx.AsyncClient(
            base_url=settings.NIA_BASE_URL,
            headers={"Authorization": f"Bearer {settings.NIA_API_KEY}"},
            timeout=10.0,
        ) as client:
            resp = await client.post("/search", json={
                "indexId": index_id,
                "query": query,
                "topK": top_k,
                "filters": filters or {},
            })
            resp.raise_for_status()
            return resp.json().get("results", [])
    except Exception:
        return []
